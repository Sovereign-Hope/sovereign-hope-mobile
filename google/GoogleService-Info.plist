<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>CLIENT_ID</key>
	<string>816081321481-ceegd9jc4qs1c9cqenck5vqur9lvulf0.apps.googleusercontent.com</string>
	<key>REVERSED_CLIENT_ID</key>
	<string>com.googleusercontent.apps.816081321481-ceegd9jc4qs1c9cqenck5vqur9lvulf0</string>
	<key>API_KEY</key>
	<string>AIzaSyBZXUNgwjdFAznceJ24t6wsaPHLco1ngn8</string>
	<key>GCM_SENDER_ID</key>
	<string>816081321481</string>
	<key>PLIST_VERSION</key>
	<string>1</string>
	<key>BUNDLE_ID</key>
	<string>com.sovereign-hope.sovereign-hope-mobile</string>
	<key>PROJECT_ID</key>
	<string>sovereign-hope-mobile</string>
	<key>STORAGE_BUCKET</key>
	<string>sovereign-hope-mobile.appspot.com</string>
	<key>IS_ADS_ENABLED</key>
	<false></false>
	<key>IS_ANALYTICS_ENABLED</key>
	<false></false>
	<key>IS_APPINVITE_ENABLED</key>
	<true></true>
	<key>IS_GCM_ENABLED</key>
	<true></true>
	<key>IS_SIGNIN_ENABLED</key>
	<true></true>
	<key>GOOGLE_APP_ID</key>
	<string>1:816081321481:ios:570b656b10b73e8d3f42d8</string>
</dict>
</plist>