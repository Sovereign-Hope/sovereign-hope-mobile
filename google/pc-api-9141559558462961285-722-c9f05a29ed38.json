{
  "type": "service_account",
  "project_id": "pc-api-9141559558462961285-722",
  "private_key_id": "c9f05a29ed381e3a0461165c9bb76fbd2ef33e48",
  "private_key": "-----BEGIN PRIVATE KEY-----\nMIIEvwIBADANBgkqhkiG9w0BAQEFAASCBKkwggSlAgEAAoIBAQC4a9+CLrNc4Uko\nTmvxoEj3JRlfZTTiktLJCQYR/DVFHoXWgOdADgZ/FJK/aXkhpHM/xMi7DAO526+o\npTb8Xx4423SselxInZkA5K1wufVclHjKhogkBcw6mHgc/4M9vie9nDfqfehPvRuL\nnRMV+u4VAT+r062WwLpI7CpE7tzdiokMj1VXB2X2je/Yy/KuHSLLH4aaVT1ceHwg\nhCwnpDyE1OuZS68cbuWPoTs/qI644t4T3FlvnO/Z84kn/SKoIjY90O8Q1kfUMmvm\nCoxu7AhGN+8y6OOy/Kqo+ozKSOviAFhZ80tEhRSzERw7qw504uWgjOlZKrZQlDA4\n85k1cJOVAgMBAAECggEAJJOQgrtMa5Fq9wmGsyd9ETdUbu0drgo8sA6QG6UP06OY\nG9D5DGU7uJyV9kyu75M0IGOhBCf/3S7LCgGqhCdBUTmFTGgAQY6wN9Mi6waWuUPh\nKRnBipNzy/NNb0Ll/qrmKt7+4A0UtCvneKswi912tsfGnweemBWlXK0qPP265F0s\ncHGO98Hbw7+LkViT5SeNcsavFFv6ULe8bADVWkgT+Yd+QTVW98u0gV9ACnL7yVNj\n+bpmcdb9BMUfmAVllAuLgeY0BsXM3DLH2zVOa/filwVsysMdjflYA6HltOX4BEvY\nqK83M96mT8reca1tcI68lt9TVXyjLX9BUb7frnreyQKBgQD7gVIQ0jg4u5e7HMf7\n3DngEzbKtJk2FP0xqIxYpe8BNSEjQeUl4fe6Uypxi8YyL/0tY6BdCT8KfM/yhaKu\n44mcxQ66XiSCK5Sc9gfPQCP/Sm5JNU9/FA/kzvrP4QRMUmdEBB45PphCTU1mBtHW\nBIlIG5uZzhBEKlG7nbh+b3eGFwKBgQC7t6H2OT3ft5uAs5VAfFeRxN21GgZEPgbe\ndutQCbwy47QaGkYxXWRLedbpp4jmjbgW7LENNwfFYiOKtzX7/e1CRY5yrgKl/RO+\nMlxwWIeKo8njb7c+tJyIxttanaT0cU8p6TH+x3RSfL7SmKrBwUtfaGzqIsIKRgoS\nRH0id50rMwKBgQC+k1OYI3MuqhdjVX/EPTAttuGB4C9oZ6ckBc+zLKlahOPDT59P\ngVax/Wrf90YLqYly51cQqHz4eXMpFlIbq5cPD9MFLWkEqflGprdsN/IO9Ft1ZcwD\nQzR38M7kzmDvZFGTnHs6bU/6jZDkiq5eObm8xg62EQsKW/bq76xpQhWTqQKBgQCM\nXh5gLiGUXomLTb43gENDH0DiIVm6Qd+FDsubCAyQyTBtrGP8medbgPoxsjNuo16a\n1db3GMVY1a/aFHHrws9dlQGVw4T7dikDOklhQPHpu9xFk84kt42twXVZZzRoDSqz\nHktiRuc2n01Z0Qsu40lnSIDY+7C5WfJCArDHy0XdfQKBgQDyLswh4gsWMFUKgm5g\nkxHKzmpO8wAFO93EzVeXOGwSxfvwkk14olQs9U4MQXM69VibJbcJJW2eLEXzdqYm\nFsUSY7+tFEdTecuCSVqKsTrqn7jA/yPy+rqYocUCGJtS/uTcqQ5QICg3fagPTmkb\n5Z6Ui0m5Sg79dZnUgrmSfxSZNQ==\n-----END PRIVATE KEY-----\n",
  "client_email": "expo-sovereign-hope-mobile@pc-api-9141559558462961285-722.iam.gserviceaccount.com",
  "client_id": "110675489499935628416",
  "auth_uri": "https://accounts.google.com/o/oauth2/auth",
  "token_uri": "https://oauth2.googleapis.com/token",
  "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
  "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/expo-sovereign-hope-mobile%40pc-api-9141559558462961285-722.iam.gserviceaccount.com"
}
