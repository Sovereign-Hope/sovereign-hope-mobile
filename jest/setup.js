/* This setup file will be fairly static and pragmatic */
import { Ionicons } from "@expo/vector-icons";
import * as Font from "expo-font";

import "react-native-gesture-handler/jestSetup";

jest.mock("react-native-reanimated", () => {
  const Reanimated = jest.requireActual("react-native-reanimated/mock");
  // The mock for `call` immediately calls the callback which is incorrect
  // So we override it with a no-op
  Reanimated.default.call = jest.fn();
  return Reanimated;
});

// Silence the warning: Animated: `useNativeDriver` is not supported because the native animated module is missing
jest.mock("react-native/Libraries/Animated/src/NativeAnimatedHelper");
jest.mock("react-native/Libraries/Animated/NativeAnimatedHelper");

jest.mock("expo-auth-session/providers/google");
jest.mock("expo-auth-session/providers/facebook");

jest.mock("@react-native-community/netinfo");

jest.mock("react-native-screens", () => {
  const RealComponent = jest.requireActual("react-native-screens");
  RealComponent.enableScreens = jest.fn();
  return RealComponent;
});

jest.mock("@react-navigation/native/lib/commonjs/useLinking.native", () => ({
  default: () => ({ getInitialState: { then: jest.fn() } }),
  __esModule: true,
}));

jest.mock("expo-notifications");

global.beforeAll(async () => Promise.all([Font.loadAsync(Ionicons.font)]));
