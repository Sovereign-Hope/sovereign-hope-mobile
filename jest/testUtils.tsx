import React from "react";
import { render as rtlRender, RenderAPI } from "@testing-library/react-native";
import { IntlProvider } from "react-intl";
import { Provider as StoreProvider } from "react-redux";
import { store } from "src/app/store";
import messages from "../compiledStrings/en.json";

function render(
  ui: React.ReactElement<any, string | React.JSXElementConstructor<any>>,
  { locale = "en", ...renderOptions } = {}
): RenderAPI {
  function Wrapper({ children }: { children: React.ReactNode }) {
    return (
      <StoreProvider store={store}>
        <IntlProvider locale={locale} messages={messages}>
          {children}
        </IntlProvider>
      </StoreProvider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything
export * from "@testing-library/react-native";

// override render method
export { render };
