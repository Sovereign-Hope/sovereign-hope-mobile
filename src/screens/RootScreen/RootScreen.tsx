import React from "react";
import { useColorScheme } from "src/hooks/useColorScheme";
import { ReadingPlanScreen } from "src/screens/ReadingPlanScreen/ReadingPlanScreen";
import { NavigationContainer } from "@react-navigation/native";
import { lightTheme, darkTheme } from "src/style/themes";
import { createNativeStackNavigator } from "@react-navigation/native-stack";
import { createBottomTabNavigator } from "@react-navigation/bottom-tabs";
import { enableScreens } from "react-native-screens";
import { Ionicons } from "@expo/vector-icons";
import { RootStackParamList } from "src/navigation/RootNavigator";
import { colors } from "src/style/colors";
import { TodayScreen } from "../TodayScreen/TodayScreen";
import { ReadScreen } from "../ReadScreen/ReadScreen";

// React Navigation configuration
enableScreens();
const Stack = createNativeStackNavigator<RootStackParamList>();
const Tab = createBottomTabNavigator<RootStackParamList>();

const HomeScreen = (): JSX.Element => (
  <Tab.Navigator>
    <Tab.Screen
      name="Today"
      component={TodayScreen}
      options={{
        headerShown: false,
        // eslint-disable-next-line react/display-name
        tabBarIcon: ({
          focused,
          color,
          size,
        }: {
          focused: boolean;
          color: string;
          size: number;
        }) => (
          <Ionicons
            name="today"
            size={size}
            color={focused ? colors.accent : color}
          />
        ),
      }}
    />
    <Tab.Screen
      name="Reading Plan"
      component={ReadingPlanScreen}
      options={{
        lazy: false,
        // eslint-disable-next-line react/display-name
        tabBarIcon: ({
          focused,
          color,
          size,
        }: {
          focused: boolean;
          color: string;
          size: number;
        }) => (
          <Ionicons
            name="ios-book"
            size={size}
            color={focused ? colors.accent : color}
          />
        ),
      }}
    />
  </Tab.Navigator>
);

export const RootScreen = (): JSX.Element => {
  const colorScheme = useColorScheme();

  return (
    <NavigationContainer
      theme={colorScheme === "dark" ? darkTheme : lightTheme}
    >
      <Stack.Navigator
        screenOptions={{
          headerTintColor: colors.accent,
          headerShadowVisible: false,
          headerTitleStyle: {
            color:
              colorScheme === "dark"
                ? darkTheme.colors.text
                : lightTheme.colors.text,
          },
        }}
      >
        <Stack.Screen
          name="Home"
          component={HomeScreen}
          options={{ headerShown: false }}
        />
        <Stack.Screen name="Read" component={ReadScreen} />
      </Stack.Navigator>
    </NavigationContainer>
  );
};
