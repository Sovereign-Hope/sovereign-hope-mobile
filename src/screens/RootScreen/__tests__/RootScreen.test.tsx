import React from "react";
import { cleanup, render, waitFor } from "jest/testUtils";
import { useNetInfo } from "@react-native-community/netinfo";
import * as Google from "expo-auth-session/providers/google";
import * as Facebook from "expo-auth-session/providers/facebook";
import * as SecureStore from "expo-secure-store";
import { RootScreen } from "../RootScreen";

jest.mock("expo-secure-store");

describe("RootScreen", () => {
  beforeEach(() => {
    (Google.useAuthRequest as jest.Mock).mockReturnValueOnce([
      jest.fn(),
      {},
      jest.fn(),
    ]);
    (Facebook.useAuthRequest as jest.Mock).mockReturnValueOnce([
      jest.fn(),
      jest.fn(),
      jest.fn(),
    ]);
    (useNetInfo as jest.Mock).mockReturnValue({
      isConnected: true,
      isInternetReachable: true,
    });
  });

  afterEach(cleanup);

  it("Shows the Landing Screen when a user token is not available", () => {
    const wrapper = render(<RootScreen />);
    expect(wrapper.getByText("Welcome!")).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });

  it("Shows the Home screen when a user token is available", async () => {
    (SecureStore.getItemAsync as jest.Mock).mockResolvedValue("accessToken");
    const wrapper = render(<RootScreen />);
    expect(await waitFor(() => wrapper.getByText("Home Screen"))).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
});
