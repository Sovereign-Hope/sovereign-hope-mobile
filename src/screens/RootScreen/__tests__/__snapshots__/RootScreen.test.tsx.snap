// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RootScreen Shows the Home screen when a user token is available 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      forwardedRef={[Function]}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="pop"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="#FC5B30"
        direction="ltr"
        hidden={false}
        hideBackButton={false}
        hideShadow={true}
        largeTitleHideShadow={false}
        title="Home"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            undefined,
          ]
        }
      >
        <View>
          <Text>
            Home Screen
          </Text>
          <View
            accessible={true}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#333333",
                "borderRadius": 10,
                "elevation": 2,
                "flexDirection": "row",
                "justifyContent": "center",
                "minHeight": 44,
                "opacity": 1,
                "paddingHorizontal": 16,
                "paddingVertical": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "flex": 1,
                  "textAlign": "center",
                }
              }
            >
              Logout
            </Text>
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;

exports[`RootScreen Shows the Landing Screen when a user token is not available 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      forwardedRef={[Function]}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="pop"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="#FC5B30"
        direction="ltr"
        hidden={true}
        hideBackButton={false}
        hideShadow={true}
        largeTitleHideShadow={false}
        title="Landing"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            undefined,
          ]
        }
      >
        <RNCSafeAreaView
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Image
              accessibilityIgnoresInvertColors={false}
              source={1}
              style={
                Object {
                  "height": "80%",
                  "marginVertical": 32,
                  "maxHeight": 250,
                  "resizeMode": "contain",
                }
              }
            />
          </View>
          <Text
            style={
              Object {
                "color": "rgb(28, 28, 30)",
                "fontSize": 36,
                "fontWeight": "600",
              }
            }
          >
            Welcome!
          </Text>
          <Text
            style={
              Object {
                "color": "rgb(28, 28, 30)",
                "fontSize": 16,
                "fontWeight": "400",
                "margin": 16,
                "textAlign": "center",
              }
            }
          >
            Sign in to your Submittable account or create a free Submittable account to get started.
          </Text>
          <View
            style={
              Object {
                "justifyContent": "flex-end",
                "padding": 16,
                "width": "100%",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#333333",
                  "borderRadius": 10,
                  "elevation": 2,
                  "flexDirection": "row",
                  "height": 44,
                  "justifyContent": "center",
                  "marginBottom": 8,
                  "minHeight": 44,
                  "opacity": 1,
                  "paddingHorizontal": 16,
                  "paddingVertical": 8,
                  "width": "100%",
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 22,
                    },
                    Object {
                      "marginRight": 4,
                    },
                    Object {
                      "fontFamily": "ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
              <Text
                style={
                  Object {
                    "color": "#FFFFFF",
                    "flex": 1,
                    "textAlign": "center",
                  }
                }
              >
                Continue with Email
              </Text>
            </View>
            <View
              accessible={true}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#333333",
                  "borderRadius": 10,
                  "elevation": 2,
                  "flexDirection": "row",
                  "height": 44,
                  "justifyContent": "center",
                  "marginBottom": 8,
                  "minHeight": 44,
                  "opacity": 1,
                  "paddingHorizontal": 16,
                  "paddingVertical": 8,
                  "width": "100%",
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 22,
                    },
                    Object {
                      "marginRight": 4,
                    },
                    Object {
                      "fontFamily": "ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
              <Text
                style={
                  Object {
                    "color": "#FFFFFF",
                    "flex": 1,
                    "textAlign": "center",
                  }
                }
              >
                Continue with Facebook
              </Text>
            </View>
            <View
              accessible={true}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#333333",
                  "borderRadius": 10,
                  "elevation": 2,
                  "flexDirection": "row",
                  "height": 44,
                  "justifyContent": "center",
                  "marginBottom": 8,
                  "minHeight": 44,
                  "opacity": 1,
                  "paddingHorizontal": 16,
                  "paddingVertical": 8,
                  "width": "100%",
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 22,
                    },
                    Object {
                      "marginRight": 4,
                    },
                    Object {
                      "fontFamily": "ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
              <Text
                style={
                  Object {
                    "color": "#FFFFFF",
                    "flex": 1,
                    "textAlign": "center",
                  }
                }
              >
                Continue with Google
              </Text>
            </View>
            <ViewManagerAdapter_ExpoAppleAuthenticationButtonContinueBlack
              proxiedProperties={
                Object {
                  "cornerRadius": 5,
                  "onButtonPress": [Function],
                }
              }
              style={
                Object {
                  "height": 44,
                  "marginBottom": 8,
                  "width": "100%",
                }
              }
            />
          </View>
        </RNCSafeAreaView>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
